### ISSUE TITLE
Segfault for custom op without inputs

### ISSUE THREAD

**Issue by aseyboldt** [apache/incubator-mxnet#8618]
Calling `bind` on custom ops that don't have inputs, but auxiliary data segfaults:

```python
import mxnet as mx
import numpy as np


class Op(mx.operator.CustomOp):
    pass


@mx.operator.register("custom_op")
class Prop(mx.operator.CustomOpProp):
    def list_arguments(self):
        return []
    
    def list_outputs(self):
        return ['out']
    
    def list_auxiliary_states(self):
        return ['value']

    def infer_shape(self, in_shape):
        return [], [(2, 2)], [(2, 2)]
    
    def infer_type(self, in_shape):
        return [], [np.float32], [np.float32]
    
    def create_operator(self, ctx, shapes, dtypes):
        return Op()
    

foo = mx.symbol.Custom(name='foo', op_type='custom_op')
data = mx.nd.ones((2, 2))

foo.bind(mx.cpu(), [], aux_states={'foo_value': data})
```

According to lldb the segfault happens in `mxnet::op::custom::CreateState`, somewhere before the call to `ostream` in L242, so in this region:

https://github.com/apache/incubator-mxnet/blob/974ac1758765291337129f546434993376eae687/src/operator/custom/custom.cc#L223-L240

My guess is that `num_args` counts only the number of arguments, but excludes the auxiliary inputs, while `in_shape` and `in_type` count all of them. If so, the assignment in L235 is illegal, because `shapes.size() == params.num_args == 0`.

**Comment by piiswrong** [comment]
Thanks for reporting.
I'm working on a fix

**Comment by aseyboldt** [comment]
Fixed in #7967.
